File: app/src/main/java/org/metabrainz/android/view/HtmlAssetTextView.java
package org.metabrainz.android.view;

import android.content.Context;
import androidx.appcompat.widget.AppCompatTextView;
import android.text.Html;
import android.util.AttributeSet;

import org.metabrainz.android.util.Utils;

public class HtmlAssetTextView extends AppCompatTextView {
    
    private Context context;
    
    public HtmlAssetTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
    }
    
    public void setAsset(String asset) {
        setText(Html.fromHtml(Utils.stringFromAsset(context, asset)));
    }

}

File: app/src/main/java/org/metabrainz/mobile/presentation/view/HtmlAssetTextView.kt
package org.metabrainz.mobile.presentation.view

import android.content.Context
import android.text.Html
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.text.HtmlCompat
import org.metabrainz.mobile.util.Utils

class HtmlAssetTextView(context: Context, attrs: AttributeSet?) : AppCompatTextView(context, attrs) {
    fun setAsset(asset: String?) {
        text = HtmlCompat.fromHtml(Utils.stringFromAsset(context, asset), HtmlCompat.FROM_HTML_MODE_LEGACY)
    }
}